app/controllers/kernel_builds_controller.rb

```diff
@@ -71,4 +68,10 @@ def set_kernel_build
     def kernel_build_params
       params.require(:kernel_build).permit(:kernel_id, :config_id, :artifact_url, kernel_configs_attributes: [:config_url], kernel_sources_attributes: [:git_repo, :git_ref])
     end
+
+    # Set collections for kernel_sources and kernel_configs
+    def set_collections
```

Comment:
This method definition is vague.  What collections? For what purpose?  Typically I would make this two different actions: `set_user_kernel_configs` and `set_user_kernel_sources` and have a before action for each.  It describes what it's doing and why we want to do it before loading the controller action.  However, if you wanted to consolidate this, something like `set_user_kernel_options` or `load_current_user_kernel_variants` is a reasonable generalization while still being descriptive.

I'm foregoing a suggestion block since it would also require a change to L4 when this is being called.

Reply:
Sounds good, I was going to go down that route but though in this context it would be unnecessary code.